File: src/v3xctrl_control/message/CommandAck.py
from .Message import Message


class CommandAck(Message):
    """Acknowledgment message for a Command."""

    def __init__(self, i: str, timestamp: float | None = None) -> None:
        super().__init__({
            "i": i
        }, timestamp)

        self.command_id = i

    def get_command_id(self) -> str:
        return self.command_id

File: android/app/src/main/java/com/v3xctrl/viewer/messages/CommandAck.kt
package com.v3xctrl.viewer.messages

import org.msgpack.core.MessageBufferPacker

/**
 * CommandAck message - acknowledgment that a command was received.
 */
class CommandAck(
    val commandId: String,
    timestamp: Double = System.currentTimeMillis() / 1000.0
) : Message(timestamp) {

    override val type: String = "CommandAck"

    companion object {
        fun fromPayload(payload: Map<String, Any>, timestamp: Double) = CommandAck(
            commandId = payload["i"] as? String ?: "",
            timestamp = timestamp
        )
    }

    override fun packPayload(packer: MessageBufferPacker) {
        packer.packMapHeader(1)
        packer.packString("i")
        packer.packString(commandId)
    }

    override fun toString(): String {
        return "CommandAck(commandId=$commandId)"
    }
}
